### DISCUSSION TITLE
clustermap: tight_layout hardocoded parameters

### DISCUSSION THREAD

**Discussion by dmaziec** [mwaskom/seaborn#3321] [Q&A] [answered]
Hello!

Thank you for your amazing package that makes data science easier!

I am using the `clustermap` method and run into problems related the color bar that overlapped the heat map. In order to increase the distance between the heat map and the color bar I would like to increase the padding that is set in `matplotlib.pyplot.tight_layout`, the `h_pad` and `w_pad` parameters. Currently, these values are hardcoded  and I believe that makes the color bar overlap one of the dendrograms sometimes:
https://github.com/mwaskom/seaborn/blob/4f7255446ed04048e46ff962eb0926351d40680d/seaborn/matrix.py#L1105

Please see my attached picture below.  When I increased the padding, it helped me and there was no overlap. 
![image](https://user-images.githubusercontent.com/52524599/229847770-10547ffa-c423-4a66-b0e4-e01a007d844e.png)

Would it be possible to have the parameters of `tight_layout` accessible through `clustermap` ? 
Otherwise, what would be the other recommended solution to solve this problem?

Thank you, 
Dominika

**Comment by mwaskom** [comment]
Can you get the result you're looking for by tweaking `cbar_pos`? (It would be easier to help with a reproducible example).

**Comment by dmaziec** [comment, marked as answer]
I am running my analysis for 40 different datasets so I would prefer a more generic, automated way instead of using `cbar_pos`. 

Here is my code example: 
Please download this dataset and run the Python script posted below from the same folder where the file is placed: [Stomach-AdenoCA.csv](https://github.com/mwaskom/seaborn/files/11225250/Stomach-AdenoCA.csv) 

The output heat map should be saved in `Stomach-AdenoCA.pdf`. 
```

import pandas
import matplotlib.pyplot as plt
import seaborn as sns

cluster_data = pandas.read_csv(
        "Stomach-AdenoCA.csv",
        index_col=False,
    )
his="Stomach-AdenoCA"
uuids = cluster_data["UUID"]
cluster_map = sns.clustermap(
                cluster_data.drop("UUID", axis=1),
                cmap=sns.color_palette("crest", as_cmap=True),  # green colors
                yticklabels=list(uuids),  # uuids labels instead of indexes
            )


cluster_map.ax_heatmap.set_yticklabels(
    cluster_map.ax_heatmap.get_yticklabels(),
    fontdict={"fontsize": 8},
)
cluster_map.ax_heatmap.set_xticklabels(
    cluster_map.ax_heatmap.get_xticklabels(), fontdict={"fontsize": 8}
)
cluster_map.ax_heatmap.set_ylabel("Sample UUID")
cluster_map.ax_heatmap.set_xlabel("Rearrangement type")
cluster_map.ax_heatmap.tick_params(axis="y", width=0.5)
cluster_map.figure.suptitle(
    f"Clustering of structural variant patterns of {his} samples.",
    size="xx-large",
    x=0.55,
    y=1,
)
cluster_map.ax_col_dendrogram.set_visible(False)

cluster_map.savefig(f"{his}.pdf")

plt.close()
```

To solve my problem I changed the parameters of `tight_layout` following the implementation in  the `clustermap` function . This is the code I reused but I changed  the value of `h_pad`. I added this piece of code before  saving the map to the file (`savefig`) : 

```
# to avoid cbar overlapping with the dendrogram
cluster_map.ax_cbar.set_axis_off()
cluster_map.figure.tight_layout(h_pad=1, w_pad=0.02, rect=[0, 0, 1, 1])
cluster_map.ax_cbar.set_axis_on()
cluster_map.ax_cbar.set_position((0.02, 0.8, 0.05, 0.18))
```
I don't struggle with this anymore since I figured how to do it on my end so feel free to close the issue.
